flytekit/core/type_engine.py

```diff
@@ -271,6 +285,9 @@ def to_python_value(self, ctx: FlyteContext, lv: Literal, expected_python_type:
                 f"Cannot convert to type {expected_python_type}, only {self._type} is supported"
             )
 
+        if lv.scalar and lv.scalar.binary:
+            return self.from_binary_idl(lv.scalar.binary, expected_python_type)  # type: ignore
```

Comment:
Should we move this to `TypeEngine.to_python_value`?

Reply:
The reason why I put here is because I want to reuse it for most transformers.

Reply:
Hi, @eapolinario just communicated with pingsu, he understands why I did this now.